ngraph/core/reference/include/ngraph/runtime/reference/roi_pooling.hpp

```diff
@@ -0,0 +1,235 @@
+                        int roi_w_end = std::round(rois[roi_idx + 3] * spatial_scale);
+                        int roi_h_end = std::round(rois[roi_idx + 4] * spatial_scale);
+
+                        // Force malformed ROIs to be 1x1
+                        int roi_height = std::max(roi_h_end - roi_h_start + 1, 1);
+                        int roi_width = std::max(roi_w_end - roi_w_start + 1, 1);
+
+                        // Divide ROIs into sub-regions for max pooling
+                        T bin_size_h = static_cast<T>(roi_height) / pooled_h;
+                        T bin_size_w = static_cast<T>(roi_width) / pooled_w;
+
+                        const T* batch_data =
+                            feature_maps + roi_batch_id * channels * height * width;
+
+                        for (unsigned int c = 0; c < channels; c++)
+                        {
+                            for (unsigned int ph = 0; ph < pooled_h; ph++)
+                            {
+                                for (unsigned int pw = 0; pw < pooled_w; pw++)
+                                {
+                                    // Compute pooling region for this output unit:
+                                    //  start (included) = floor(ph * roi_height / pooled_h)
+                                    //  end (excluded) = ceil((ph + 1) * roi_height / pooled_h)
+                                    int h_start = static_cast<int>(
+                                        std::floor(static_cast<T>(ph) * bin_size_h));
+                                    int w_start = static_cast<int>(
+                                        std::floor(static_cast<T>(pw) * bin_size_w));
+                                    int h_end = static_cast<int>(
+                                        std::ceil(static_cast<T>(ph + 1) * bin_size_h));
+                                    int w_end = static_cast<int>(
+                                        std::ceil(static_cast<T>(pw + 1) * bin_size_w));
+
+                                    // Add ROI offsets and clip to input boundaries
+                                    h_start = std::min(std::max(h_start + roi_h_start, 0), height);
+                                    w_start = std::min(std::max(w_start + roi_w_start, 0), width);
+                                    h_end = std::min(std::max(h_end + roi_h_start, 0), height);
+                                    w_end = std::min(std::max(w_end + roi_w_start, 0), width);
+
+                                    const size_t pool_index =
+                                        roi_num * channels * pooled_h * pooled_w +
+                                        c * pooled_h * pooled_w + ph * pooled_w + pw;
+
+                                    // Define an empty pooling region to be zero
+                                    bool is_empty = (h_end <= h_start) || (w_end <= w_start);
+                                    if (is_empty)
```

Comment:
```suggestion
                                    if (h_end <= h_start || w_end <= w_start)
```

Reply:
@ilyachur , Why we can't use temporary boolean variables?

Reply:
We use this variable only in one place.
And why do we need to add a new variable if we can do it in-place?

Reply:
To give a meaningful name to this OR condition. I personally like it.

Reply:
No difference in asm: https://godbolt.org/z/vG7rb3